### ISSUE TITLE
./install.sh --all-source fails to build

### ISSUE THREAD

**Issue by zeeduino** [hioa-cs/IncludeOS#860]
An all-source build fails on fresh install of Ubuntu 14.04 and Ubuntu 16.04. It fails when building libgcc and there seems to be multiple errors present:
- `xgcc: error: unrecognized command line option '-V'`
- `xgcc: error: unrecognized command line option '-qversion'; did you mean '--version'?`
- `/home/user/IncludeOS_build/build_gcc/./gcc/as: 106: exec: -o: not found` (this one 2x)

I have attached part of the build output and `~/IncludeOS_build/build_gcc/i686-elf/libgcc/config.log` in attached txt file.

[source-build-error.txt](https://github.com/hioa-cs/IncludeOS/files/516919/source-build-error.txt)

I was following instructions from here https://github.com/hioa-cs/IncludeOS/wiki/Ubuntu#b-completely-build-everything-from-source-slow, and spent and hour or so scaling and spelunking the Internet trying to figure out what is the issue, but didn't find anything of help so far.

**Comment by gurka** [comment]
I just tried this on tag v0.9.2 and got this error:

```
--2016-10-10 17:14:29--  ftp://ftp.uninett.no/pub/gnu/binutils/binutils-2.26.tar.gz
           => ‘binutils-2.26.tar.gz’
Resolving ftp.uninett.no (ftp.uninett.no)... 128.39.3.170, 2001:700:0:12e::f70
Connecting to ftp.uninett.no (ftp.uninett.no)|128.39.3.170|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/gnu/binutils ...
No such directory ‘pub/gnu/binutils’.
```

I abort the build after this error. Gonna try to find an alternative location for binutils-2.26 and see if I encounter any other errors.

@zeeduino What branch/tag were you trying on? Latest master?

**Comment by gurka** [comment]
Now it worked much better. Well, yeah until `[30140.140597] Out of memory: Kill process 10414 (clang) score 14 or sacrifice child` ...

@zeeduino Could you try just changing the url in `etc/build_binutils.sh` to `ftp://ftp.gnu.org/gnu/binutils/binutils-$binutils_version.tar.gz`?

```
diff --git a/etc/build_binutils.sh b/etc/build_binutils.sh
index 5de084c..62f3e2a 100755
--- a/etc/build_binutils.sh
+++ b/etc/build_binutils.sh
@@ -2,7 +2,8 @@ pushd $BUILD_DIR

 if [ ! -f binutils-$binutils_version.tar.gz ]; then
     echo -e "\n\n >>> Getting binutils into `pwd` \n"
-    wget -c --trust-server-name ftp://ftp.uninett.no/pub/gnu/binutils/binutils-$binutils_version.tar.gz
+    wget -c --trust-server-name ftp://ftp.gnu.org/gnu/binutils/binutils-$binutils_version.tar.gz
 fi
```

**Comment by alfred-bratterud** [comment]
I can confirm this error. @gurka - your solution seems to work, would you mind sending a PR?

**Comment by ingve** [comment]
@zeeduino, did the fix in #870 get the build to complete for you?

**Comment by zeeduino** [comment]
I completely dropped off the radar there for a while...

Thanks for quick response and issue resolution!!

@ingve Now, all-source build works from master on Ubuntu 14.04.

However, running

    ./test.sh

does not, it complains that `objcopy` doesn't recognize option `--update-section`. The reason why is that Ubuntu 14.04 ships with binutils 2.24, and `--update-section` is introduced in 2.26.

So I did a quick copy/paste from `install_all_source.sh`:

    export BUILD_DIR=$HOME/IncludeOS_build
    export TEMP_INSTALL_DIR=$BUILD_DIR/IncludeOS_TEMP_install
    export PREFIX=$TEMP_INSTALL_DIR
    export PATH="$PREFIX/bin:$PATH"

And then:

    export OBJCOPY_INC=i686-elf-objcopy
    export STRIP_INC=i686-elf-strip

which did the trick and `./test.sh` ran and passed with flying colors.


As a suggestion, since binutils have been built from source, wouldn't it be a good idea to have binaries copied over to `$INSTALL_DIR` and use those in scripts later on, instead of relying on OS supplied binutils?
